experimental/experimental.go

```diff
@@ -2,3 +2,15 @@
 //
 // Note: All features here may be changed or deleted at any time, so use with caution!
 package experimental
+
+import (
+	"github.com/tetratelabs/wazero/api"
+)
+
+// InternalModule exposes additional module information not available through
+// api.Module.
+type InternalModule interface {
+	// ViewGlobals return a copy of all the globals defined in the module. It
+	// includes non-exported globals.
+	ViewGlobals() []api.Global
```

Comment:
If we limit this to by-index-access only, then we should be able to avoid allocating `globalsView []api.Global`?
```suggestion
	Global(index int) []api.Global
	GlobalsCount() int
```